code/ATMOSPHERICS/components/components_base.dm

```diff
@@ -199,7 +195,6 @@ Helpers
 */
 
 /obj/machinery/atmospherics/components/proc/update_parents()
-	for(var/P in parents)
-		var/datum/pipeline/parent = parents["[P]"]
-		if (parent && istype(parent))
-			parent.update = 1
\ No newline at end of file
+	for(var/I = 1; I <= device_type; I++)
+		var/datum/pipeline/parent = parents["p[I]"]
+		parent.update = 1
```

Comment:
its pretty good at runtiming like hell when there's a bug, so it can stay temporally(TM)
plus consistency with the rest of these procs under this section